text/0005-media-webrtc-sdk.md

```diff
@@ -0,0 +1,589 @@
+
+# 3. User Benefit
+
+User can use WebRTC to connect to the media server and create complex media stream topologies.
+
+# 4. Design Proposal
+
+To ensure simplicity, we propose a SDK protocol that exclusively uses HTTP (not WebSocket) and WebRTC.
+
+- HTTP is used only for sending RPC requests to the cluster, typically during the Connect and Retry phases.
+- WebRTC is utilized for sending and receiving media streams, as well as RPC and event communication after establishing a connection.
+
+## 4.1 HTTP Request/Response format
+
+
+**Request and Response Format**
+
+All requests and responses will be encoded in JSON format. The format is described as follows:
+
+**_Body:_**: JSON
+
+**_Response:_**
+
+```
+{
+    success: bool,
+    error_code: Option<String>,
+    error_msg: Option<String>,
+    data: Option<JSON>,
+}
+```
+
+
+## 4.2 Connect Establishment
+
+Before connecting to the server, the client needs to prepare the following:
+
+- Enable WebRTC connection with data channel.
+- Create a list of senders and receivers.
+- Generate WebRTC OfferSDP.
+
+Once the client is ready, it can send a connect request to the server.
+
+**_Endpoint_**: `POST GATEWAY/webrtc/connect`
+
+**_Headers_**: `Authorization: Bear {token}`
+
+**_Body_**:
+
+```
+{
+    version: Option<String>,
+    room: String,
+    peer: String,
+    event: {
+        publish: "full" | "track",
+        subscribe: "full" | "track" | "manual",
+    },
+    bitrate: {
+        ingress: "save" | "max",
+    },
+    features: JSON,
+    tracks: {
+        receivers: [
+            {
+                kind: "audio" | "video",
+                id: String,
+                state: Option<{
+                    remote: Option<{
+                        peer: String,
+                        track: String,
+                    }>,
+                    limit: Option<{
+                        priority: u16,
+                        min_spatial: Option<u8>,
+                        max_spatial: u8,
+                        min_temporal: Option<u8>,
+                        max_temporal: u8,
+                    }>,
+                }>
+            }
+        ],
+        senders: [
+            {
+                kind: "audio" | "video",
+                id: String,
+                uuid: String,
+                label: String,
+                state: Option<{
+                    screen: bool,
+                    pause: bool,
+                }>,
+            }
+        ],
+    },
+    sdp: Option<String>
+}
+```
+
+**_Response Data:_**
+
+```
+{
+    sdp: Option<String>,
+    conn_id: String,
+}
+```
+
+The explanation of each request parameter:
+
+- version: is the version of the client SDK.
+- event:
+
+    - publish: `full` will publish both peer info and track info. `track` will only publish track info.
```

Comment:
Can you elaborate on `peer info` and `track info`?

Reply:
What do you mean?

Reply:
can you describe more clearly on what peer info and track info will contain

Reply:
I just added the missing peer and track events. Can you please check if there are any inconsistencies?